src/peep/peep.c

```diff
@@ -6160,14 +6171,15 @@ void peep_set_map_tooltip(rct_peep *peep)
 
 
 void sub_693BAB(rct_peep* peep) {
-	uint8 bl = peep->var_6F;
-	if (bl != peep->action_sprite_type) {
+	// Add spriteType as function parameter and make peep->next_action_sprite_type obsolete?
+	uint8 spriteType = peep->next_action_sprite_type;
+	if (spriteType != peep->action_sprite_type) {
 		invalidate_sprite_2((rct_sprite*)peep);
-		peep->action_sprite_type = bl;
+		peep->action_sprite_type = spriteType;
 		uint8* edx = RCT2_ADDRESS(0x98270C, uint8*)[peep->sprite_type * 2];
```

Comment:
Okay, figured it out I think. The structs need to be defined as follows:
```
typedef struct {
	uint8 sprite_width;             // 0x00
	uint8 sprite_height_negative;   // 0x01
	uint8 sprite_height_positive;   // 0x02
	uint8 unused;                   // 0x03
} rct_sprite_unk2;

typedef struct {
	uint32 base_image_id;
	rct_sprite_unk2* spriteb;
} rct_sprite_unk1;
```

and `0x00982708` is the start of an rct_sprite_unk1 array.